src/read.c

```diff
@@ -4247,7 +4247,7 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
         *isAlphaItemInInput = AVIF_FALSE;
         return AVIF_RESULT_OK;
     }
-    uint32_t * alphaItemIndices = avifAlloc(colorItemCount * sizeof(uint32_t));
+    uint32_t * alphaItemIndices = avifAlloc(meta->items.count * sizeof(uint32_t));
```

Comment:
an alternative solution is to check for `alphaItemCount` not exceeding `colorItemCount` in line 4262. but that involves repeating the code for the failure return. so i chose the simpler approach. the overhead should be minimal since this is an extreme case anyways.

Reply:
The correctness of this solution requires all the items in the `meta->items` array have distinct IDs. I believe that is the case, because the array is constructed by `avifMetaFindOrCreateItem()`. Please double check.